src/transforms/log_to_metric.rs

```diff
@@ -268,9 +268,11 @@ fn to_metric(config: &MetricConfig, event: &Event) -> Result<Metric, TransformEr
         .metadata()
         .clone()
         .with_schema_definition(&Arc::new(Definition::any()))
-        .with_origin_metadata(
-            DatadogMetricOriginMetadata::default().with_service(ORIGIN_SERVICE_VALUE),
-        );
+        .with_origin_metadata(DatadogMetricOriginMetadata::new(
+            None,
+            None,
+            Some(ORIGIN_SERVICE_VALUE),
+        ));
```

Comment:
This is a detail covered in the design spec in further detail but essentially in order to set the correct service value in the `datadog_metrics` sink, we'd need to know that the metric was originated from the `log_to_metric` transform specifically.
The only way I could think to do that off the top of my head (without doing it the current way) , would be to add some kind of field to the EventMetadata similar to the `source_type` , that would somehow carry the information that the `log_to_metric` transform was involved.
That seems more obscure than the current way. This transform is the only place where we kind of violate the general rule that the three fields should be set together :/
I could see it being an argument to keep the `with_service()` function and use that here instead of passing in two `None` arguments. WDYT?

There is a note about it here as well: https://github.com/vectordotdev/vector/blob/54b54a5bd8c8a6ac0f13ee2cb607e2debd0befeb/src/sinks/datadog/metrics/encoder.rs#L626-L627

Reply:
I have no strong preference for the builder form over this optional form when everything but one place uses a common pattern.